File: cropaway/cropaway/Models/CropMode.swift
//
//  CropMode.swift
//  cropaway
//

import Foundation

enum CropMode: String, Codable, CaseIterable, Identifiable {
    case rectangle = "rectangle"
    case circle = "circle"
    case freehand = "freehand"
    case ai = "ai"

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .rectangle: return "Rectangle"
        case .circle: return "Circle"
        case .freehand: return "Custom Mask"
        case .ai: return "AI Mask"
        }
    }

    var iconName: String {
        switch self {
        case .rectangle: return "rectangle"
        case .circle: return "circle"
        case .freehand: return "scribble.variable"
        case .ai: return "wand.and.stars"
        }
    }
}

File: CropawayWindows/CropawayWindows/Models/CropMode.cs
// CropMode.cs
// CropawayWindows

namespace CropawayWindows.Models;

/// <summary>
/// Available crop modes for video cropping operations.
/// </summary>
public enum CropMode
{
    Rectangle,
    Circle,
    Freehand,
    AI
}

/// <summary>
/// Extension methods for <see cref="CropMode"/> providing display names
/// and Segoe Fluent Icons glyph strings.
/// </summary>
public static class CropModeExtensions
{
    /// <summary>
    /// Gets the human-readable display name for the crop mode.
    /// </summary>
    public static string DisplayName(this CropMode mode) => mode switch
    {
        CropMode.Rectangle => "Rectangle",
        CropMode.Circle => "Circle",
        CropMode.Freehand => "Custom Mask",
        CropMode.AI => "AI Mask",
        _ => mode.ToString()
    };

    /// <summary>
    /// Gets the Segoe Fluent Icons glyph string for the crop mode.
    /// </summary>
    public static string IconName(this CropMode mode) => mode switch
    {
        CropMode.Rectangle => "\uE003",  // RectangleShape
        CropMode.Circle => "\uEA3A",     // CircleShape
        CropMode.Freehand => "\uEE56",   // Inking / Pen
        CropMode.AI => "\uE945",         // Wand / Magic
        _ => "\uE003"
    };

    /// <summary>
    /// Gets the keyboard shortcut description for the crop mode.
    /// </summary>
    public static string KeyboardShortcut(this CropMode mode) => mode switch
    {
        CropMode.Rectangle => "Ctrl+1",
        CropMode.Circle => "Ctrl+2",
        CropMode.Freehand => "Ctrl+3",
        CropMode.AI => "Ctrl+4",
        _ => string.Empty
    };
}
